core/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -215,6 +215,44 @@ impl StableXOrderBookReading for UpdatingOrderbook {
             .boxed()
     }
     fn initialize<'a>(&'a self) -> BoxFuture<'a, Result<()>> {
-        self.do_with_context(|_| async { Ok(()) }.boxed()).boxed()
+        self.do_with_context(|_| immediate!(Ok(()))).boxed()
+    }
+}
+
+impl HistoricOrderbookReading for UpdatingOrderbook {
+    fn get_auction_data_for_block<'a>(
+        &'a self,
+        block_number: BlockNumber,
+    ) -> BoxFuture<'a, Result<(AccountState, Vec<Order>)>> {
+        // NOTE: Get the timestamp for the block, this will give more accurate
+        // results as to what batch the orderbook is retrieved for.
+        self.do_with_context(move |context| {
```

Comment:
We can solve this by exposing the block number -> timestamp conversion in EventRegistry and using this from here. That way no changes to the Cache are needed.
EventRegistry will give us the timestamp of the closest lower block it has which is enough since that means it doesn't have any events for other blocks up to this number anyway.

Reply:
This was done intentionally, since we want a more accurate timestamp to use for determining the current batch ID. The reason for this is, lets say that there are no events for some blocks and we request the orderbook at a block in the middle. It could be that the block is in a different batch than the what the current batch was at the block where the last event was received.